webapp/src/components/systemConsole/questions/questions.tsx

```diff
@@ -0,0 +1,147 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React, {useCallback, useEffect, useMemo, useState} from 'react';
+import './style.scss';
+import {useDebouncedCallback} from 'use-debounce';
+import utils from 'utils/utils';
+
+import type {CustomSettingChildComponentProp} from 'components/systemConsole/index';
+
+import type {SurveyQuestionsConfig} from 'types/mattermost-webapp';
+
+export type QuestionType = 'linear_scale' | 'text';
+
+const questionTypeDisplayName = new Map<QuestionType, string>([
+    ['linear_scale', 'Linear scale question (1 to 10)'],
+    ['text', 'Textual question'],
+]);
+
+export type Question = {
+    id: string;
+    title?: string;
+    text?: string;
+    type: QuestionType;
+    system: boolean;
+    mandatory: boolean;
+    helpText?: string;
+};
+
+function SurveyQuestions({id, setSaveNeeded, onChange, config, setInitialSetting}: CustomSettingChildComponentProp) {
+    const [questions, setQuestions] = useState<Question[]>([]);
+
+    const generateDefaultQuestions = (): Question[] => {
+        return [
+            {
+                id: utils.uuid(),
+                title: 'Survey message text',
+                text: 'Please take a few moments to help us improve your experience with Mattermost.',
+                type: 'text',
+                system: false,
+                mandatory: true,
+                helpText: 'This text will be sent in the bot message preceding the survey.',
+            },
+            {
+                id: utils.uuid(),
+                text: 'How likely are you to recommend Mattermost?',
+                type: 'linear_scale',
+                system: true,
+                mandatory: true,
+            },
+            {
+                id: utils.uuid(),
+                text: 'How can we make your experience better?',
+                type: 'text',
+                system: true,
+                mandatory: true,
+            },
+            {
+                id: utils.uuid(),
+                type: 'text',
+                system: false,
+                mandatory: false,
+            },
+        ];
+    };
```

Comment:
I don't see any reason to have this function inside the component (and being regenerated on every render).

I have mixed feelings on whether it needs to be a function at all. The only reason I see is that we may want to have the ids different every time we call this, but not sure if that is actually needed.

Reply:
generating different IDs was part of the reason. While I don't know what problem it causes if IDs weren't re-generated everytime we needed this, I had a feeling it might cause issues. Its a function because if its an array inside the component, it will be generated on every render. It can either be a useMemo, or a function like this.